sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsGdsStore.scala

```diff
@@ -82,6 +82,7 @@ class RapidsGdsStore(
       closeOnExcept(DeviceMemoryBuffer.allocate(size)) { buffer =>
         CuFile.readFileToDeviceBuffer(buffer, path, fileOffset)
         logDebug(s"Created device buffer for $path $fileOffset:$size via GDS")
+        buffer.incRefCount()
```

Comment:
`DeviceMemoryBuffer.allocate` returns a buffer that has a refcount of 1, and we don't close it before it's returned.  Incrementing it here seems like we're asking it to leak, since I would only expect the caller of this method to close it once.

Reply:
Ah ok, I think the problem is we are closing the device buffer in `getDeviceMemoryBuffer`. Should be fixed now.